perllib/FixMyStreet/App/Controller/Claims.pm

```diff
@@ -0,0 +1,594 @@
+    if ($c->get_param('goto-address_unknown')) {
+        $c->set_param('goto', 'address_unknown');
+        $c->set_param('process', '');
+    }
+
+    my $form = load_form($c);
+    if ($c->get_param('process') && !$c->stash->{override_no_process}) {
+        $c->forward('/auth/check_csrf_token');
+        $form->process(params => $c->req->body_params);
+        if ($form->validated) {
+            $form = load_form($c, $form);
+        }
+    }
+
+    $form->process unless $form->processed;
+
+    $c->stash->{template} = $form->template || 'claims/index.html';
+    $c->stash->{form} = $form;
+}
+
+sub get_page : Private {
+    my ($self, $c) = @_;
+
+    my $goto = $c->get_param('goto') || '';
+    my $process = $c->get_param('process') || '';
+    $goto = 'intro' unless $goto || $process;
+    if ($goto && $process) {
+        $c->detach('/page_error_400_bad_request', [ 'Bad request' ]);
+    }
+
+    return $goto || $process;
+}
+
+sub fields : Private {
+    my ($self, $what, $fault_fixed) = @_;
+
+
+    return [
+        {
+            stage => 'start',
+            title => "About claim",
+            fields => [
+                {
+                    desc => "What are you claiming for?",
+                    value => 'what'
+                },
+                {
+                    block => 'yes_no',
+                    desc => "Claimed before",
+                    value => 'claimed_before'
+                }
+            ]
+        },
+        {
+            stage => 'about_you',
+            title => "About you",
+            fields => [
+                {
+                    desc => "Name",
+                    value => 'name'
+                },
+                {
+                    desc => "Email",
+                    value => 'email'
+                },
+                {
+                    desc => "Phone",
+                    value => 'phone'
+                },
+                {
+                    desc => "Address",
+                    value => 'address'
+                },
+            ]
+        },
+        {
+            stage => 'about_fault',
+            title => "About the fault",
+            fields => [
+                {
+                    desc => "Fixed",
+                    block => "yes_no",
+                    value => "fault_fixed",
+                },
+                {
+                    desc => "Fault reported?",
+                    hide => ($fault_fixed == 1),
+                    value => 'fault_reported'
+                },
+                {
+                    desc => "Fault ID",
+                    hide => ($fault_fixed == 1),
+                    value => 'report_id'
+                },
+            ]
+        },
+        {
+            stage => 'when',
+            title => "When did the incident happen?",
+            fields => [
+                {
+                    desc => "Date",
+                    value => 'incident_date'
```

Comment:
Error message on this for too early a date is `'2015' is not a valid value` which is not very helpful, as it doesn't say what is.